oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/listener/TraceSegmentSampler.java

```diff
@@ -25,7 +25,7 @@
  * ignored
  */
 public class TraceSegmentSampler {
-    private TraceServiceModuleConfig config;
+    private volatile TraceServiceModuleConfig config;
```

Comment:
This change is not right. Please read my review more carefully.

Reply:
Dynamic configuration will not modify TraceServiceModuleConfig#sampleRate，but modify TraceSampleRateWatcher#sampleRate

Reply:
No one will change this reference, @killGC

Reply:
okay，but this reference #traceSampleRateWatcher#sampleRate will change，to get newest sampleRate's value by this reference，no need for this reference add volatile？

Reply:
You are always using the same `config#getTraceSampleRateWatcher ` reference, right? As the `config` is never changed, you don't need this. `volatile` could cause more CPU, please use it if necessary.

Reply:
okay,i see